Added new Selenium tests

shinair opened the pull request:
Added new Selenium tests to test the following features:
1. Login
2. Logout
3. Patient Search
4. Vitals

sunbiz reviewed (commented):
Please add .pyc to the gitignore and remove spaces at the end of vitals.py

sunbiz commented on oemr-selenium/test_utils.py:
```diff
@@ -14,3 +16,29 @@ def get_user():
 
 def get_pass():
```
please remove existing functions, if you are replacing it with other functions in utls.

shinair replied:
Removed existing functions.

sunbiz commented on oemr-selenium/login_tests.py:
```diff
@@ -0,0 +1,39 @@
+        self.browser.find_element(By.ID, "clearPass").send_keys(password)
+        self.browser.find_element(By.ID, "login-button").submit()
+
+        assert "https://in-info-web18.luddy.iupui.edu/interface/main/tabs/main.php" in self.browser.current_url
```
Don't hardcode URLs in assert. Check from the URL property

shinair replied:
Modified the secret.json structure to include this value since it is the same for all the tests. Added a new utility function to read this from secret.json.

sunbiz commented on oemr-selenium/logout.py:
```diff
@@ -0,0 +1,47 @@
+        self.browser.find_element(By.ID, "clearPass").send_keys(password)
+        self.browser.find_element(By.ID, "login-button").submit()
+
+        assert "https://in-info-web18.luddy.iupui.edu/interface/main/tabs/main.php" in self.browser.current_url
```
Dont hardcode

shinair replied:
Modified the secret.json structure to include this value since it is the same for all the tests. Added a new utility function to read this from secret.json.

sunbiz commented:
I suggest adding the login/logout as separate functions in the utils that can be reused in the setup and teardown process.

shinair commented:
Made the following changes:
1. Added a new function for login in utils. Changed all the tests to use this. This function does the login part and closes all the open tabs since this was also repeated in all the tests.
2. Changed all the hard-coded urls with variables or values from secret.json.
3. Modified the readme to reflect the new secret.json structure.

sunbiz commented:
Looks good... please squash the 3 commits into 1.

shinair commented:
Done, squashed all the commits into 1.
